Core/Core/Grades/GradeListViewController.swift

```diff
@@ -0,0 +1,248 @@
+        errorView.retryButton.addTarget(self, action: #selector(refresh), for: .primaryActionTriggered)
+
+        gradingPeriodView.isHidden = true
+
+        loadingView.color = nil
+        refreshControl.color = nil
+
+        refreshControl.addTarget(self, action: #selector(refresh), for: .primaryActionTriggered)
+        tableView.refreshControl = refreshControl
+        tableView.registerHeaderFooterView(SectionHeaderView.self)
+        tableView.separatorColor = .named(.borderMedium)
+        tableView.tableHeaderView?.sizeToFit()
+
+        totalGradeHeadingLabel.text = NSLocalizedString("Total Grade", bundle: .core, comment: "")
+
+        colors.refresh()
+        courses.refresh()
+        gradingPeriods.refresh()
+        assignments.refresh()
+    }
+
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        if let index = tableView.indexPathForSelectedRow {
+            tableView.deselectRow(at: index, animated: animated)
+        }
+        if let color = color {
+            navigationController?.navigationBar.useContextColor(color)
+        }
+    }
+
+    @objc func refresh() {
+        colors.refresh(force: true)
+        courses.refresh(force: true)
+        gradingPeriods.refresh(force: true)
+        assignments.refresh(force: true) { [weak self] _ in
+            self?.refreshControl.endRefreshing()
+        }
+    }
+
+    func updateNavBar() {
+        guard let course = courses.first else { return }
+        color = colorDelegate?.iconColor ?? course.color.ensureContrast(against: .named(.white))
+        titleSubtitleView.subtitle = course.name
+        navigationController?.navigationBar.useContextColor(color)
+        view.tintColor = color
+    }
+
+    func update() {
+        if !gradingPeriodLoaded, gradingPeriodID != enrollment?.currentGradingPeriodID {
+            updateGradingPeriod(id: enrollment?.currentGradingPeriodID)
+        }
+        gradingPeriodLoaded = gradingPeriodLoaded || (courses.requested && !courses.pending)
+        gradingPeriodView.isHidden = !gradingPeriodLoaded || enrollment?.multipleGradingPeriodsEnabled != true
+        gradingPeriodLabel.text = gradingPeriodID == nil && gradingPeriodLoaded
+            ? NSLocalizedString("All Grading Periods", bundle: .core, comment: "")
+            : gradingPeriods.first(where: { $0.id == gradingPeriodID })?.title
+        let filter = gradingPeriodID == nil
+            ? NSLocalizedString("Filter", bundle: .core, comment: "")
+            : NSLocalizedString("Clear filter", bundle: .core, comment: "")
+        filterButton.setTitle(filter, for: .normal)
+
+        totalGradeLabel.text = courses.first?.hideFinalGrades == true
+            ? NSLocalizedString("N/A", bundle: .core, comment: "")
+            : enrollment?.formattedCurrentScore(gradingPeriodID: gradingPeriodID)
```

Comment:
so the course endpoint doesn't return all the enrollments? If that's so I can add an enrollments use case instead of relying on getting it from course.